src/compiler/transformers/declarations/emitResolver.ts

```diff
@@ -393,3 +614,50 @@ export function createEmitDeclarationResolver(file: SourceFile): CoreEmitResolve
         return false;
     }
 }
+
+function getSymbolName(name: Exclude<PropertyName, ComputedPropertyName>): __String;
+function getSymbolName(name: ElementAccessExpression | PropertyName): __String | undefined;
+function getSymbolName(name: ElementAccessExpression | PropertyName): __String | undefined {
+    const staticName = isPropertyName(name) ? getPropertyNameForPropertyNameNode(name) :
+        isElementAccessExpression(name) && isPropertyName(name.argumentExpression) ? getPropertyNameForPropertyNameNode(name.argumentExpression) :
+        undefined;
+
+    if (staticName) return staticName;
+
+    return getDynamicSymbolName(name);
+}
+
+function getEntityNameComponent(identifier: MemberName): __String {
+    const name = getPropertyNameForPropertyNameNode(identifier);
+    if (name && name.indexOf(".")) {
+        return name.replace(/\./g, "..") as __String;
+    }
+    return name;
+}
+
+function getDynamicSymbolName(name: ElementAccessExpression | PropertyName) {

```

Comment:
I wonder if you can use `tryGetPropertyAccessOrIdentifierToString` or `entityNameToString`.

Reply:
Switched over to `entityNameToString`